langroid/agent/special/retriever_agent.py

```diff
@@ -204,7 +216,10 @@ def llm_select_relevant_docs(
         )
         if response.message == NO_ANSWER:
             return [default_response]
-        ids = response.message.split()
+        if response.message == "NO MATCHING":
+            return [no_matching]
+        ids_lst = response.message.split('"')[1]
+        ids = ast.literal_eval(ids_lst)
```

Comment:
what happens if ids are numerical, like `[12, 1, 15]`, then `liter_eval` will produce a list of ints, but `get_documents_by_ids` expects a list of `str`

Reply:
If I understood correctly, the concern is that when the parameter passed to `liter_eval` is numerical. But `ids_lst` is always str, and `liter_eval` job is to convert `ids_lst` to a list of str. 
However, if `ids_lst` is numerical, `liter_eval` will not produce anything, instead this will raise an error.
<img width="896" alt="image" src="https://github.com/langroid/langroid/assets/15859139/12ea5173-ff00-40c1-885d-b2a618aca6b0">